docs/api-reference/test-utils/browser-driver.md

```diff
@@ -6,115 +6,108 @@
-* Opens a browser page with a URL that loads the bundle into the browser and runs the tests.
-* Extracts a pass/fail value from the browser back into node
-* Closes browser, server and terminates the current node script.
-* Passes the pass/fail value as exit status (0/1) back to the invoking shell.
+* Launch/close a Chromium browser instance
+* Start/stop a local web service.
+* Opens a browser page with a URL in the browser.
 
+To use this class, [puppeteer](https://www.npmjs.com/package/puppeteer) must be installed as a dev dependency.
 
 ## Usage
 
 ```js
 import {BrowserDriver} from 'probe.gl/test-utils';
-new BrowserDriver({...});
+new BrowserDriver({id: 'browser-test'});
 ```
 
 
-## Methods
+## Constructor
 
-### constructor
+```js
+const browserDriver = new BrowserDriver(opts);
+```
 
-`new BrowserDriver({...})`
+Parameters:
 
-### startBrowser
+* `opts` (Object)
+  - `id` (String) - an id for this `BrowserDriver` instance. Default `browser-driver`.
 
-`driver.startBrowser(options)`
 
-Returns a `Promise` that resolves when the browser has started.
+## Methods
 
+### startBrowser(options)
 
-### newPage
+Launch a new browser instance.
 
-Opens a new tab in the browser
+Parameters:
 
-`driver.newPage({url, width, height})`
+* `options` (Object) - options to pass to [puppeteer.launch](https://github.com/GoogleChrome/puppeteer/blob/v1.11.0/docs/api.md#puppeteerlaunchoptions).
 
-* `url` = `http://localhost:8080` - The url to load in the page.
-* `width` = `1550` - The width of the page
-* `height` = `850` - The height of the page
-Returns a `Promise` that resolves when the page is open.
+Returns a `Promise` that resolves when the browser has started.
 
+### openPage(options)
 
-### exposeFunction
+Open a new tab in the browser. Only works after a browser instance is started:
 
-Exposes a function on the `window` object in the controlled browser and waits until the controlled script has called that function. In addition, a string parameter to the function will be returned (as the value of the resolved `Promise`).
+```js
+browserDriver.startBrowser().openPage({url: 'http://localhost'});
+```
 
-`driver.exposeFunction(functionToExpose)`
+Parameters:
 
-* `functionToExpose`=`sendMessage` (`String`)
-Returns: a `Promise` that resolves when the browser code calls the function to the string the browser script passed to the exposed function.
+* `options` (Object)
+  - `url` (String) - The url to load in the page. Default `http://localhost`.
+  - `exposeFunctions` (Object) - keys are function names to be added to the page's `window` object, and the values are callback functions to execute in Node.js. See [exposeFunction](https://github.com/GoogleChrome/puppeteer/blob/v1.11.0/docs/api.md#pageexposefunctionname-puppeteerfunction) for details.
+  - `onLoad` (Function) - callback when the page is loaded
+  - `onConsole` (Function) - callback when the page logs to console
+  - `onError` (Function) - callback when the page crashes
```

Comment:
Nit: Sounds dramatic: "when the page crashes" ? => "called by puppeteer if the page crashes"?